### PR TITLE
Move all data source endpoints to [dsId] instead of [name]

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Move all data source related endpoint to [dsId] instead of [name] requiring changing the re.query extraction.
- Changes a few call sites to start using dataSource.sId instead of dataSource.name.
- Replace call to `getDataSource` to direct call to the `DataSourceResource` to depreate `getDataSource` and get better tracking of use of name/sId

## Risk

Low in theory but somewhat high given the number of files it touches

Testing:
- [x] Create a Folder
- [x] Upsert by product
- [x] Upsert by API
- [x] Create a Website
- [x] Search
- [x] Delete Folder
- [x] Delete Website
- [x] Create a Connection
- [x] Use in one assistant
- [x] poke operations

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/components/vaults/VaultWebsiteModal.tsx [hunk 1]
```diff
@@ -180,13 +177,7 @@ export default function VaultWebsiteModal({
...
 
     // Validate Name
-    const nameExists = dataSources.some(
-      // TODO(DATASOURCE_SID): This line was kind of buggy because dataSourceId and sId was name but
-      // dataSourceId was not defined when it mattered (creation) and this not used during patching.
-      // (currently replacing back to .name the real real value for now showing that the line makes
-      // no sense). Will move to sId in subsequent PR.
-      (d) => d.name === dataSourceName && d.name !== dsName
-    );
+    const nameExists = dataSources.some((d) => d.name === dataSourceName);
```
Comment (philipperolet) [inline front/components/vaults/VaultWebsiteModal.tsx:180]: not sure I get the change of logic here --- before, nameExists was false if the name was the current datasourceView name; now it's true in that case (and will err below with `nameError`). Is that expected?

File: front/lib/resources/data_source_resource.ts [hunk 2]
```diff
@@ -42,13 +42,38 @@ export type FetchDataSourceOrigin =
...
+  | "poke_data_sources_config"
+  | "poke_data_sources_documents"
+  | "poke_data_sources_permissions"
   | "registry_lookup"
   | "data_source_get_or_post"
   | "data_source_managed_update"
   | "vault_data_source_config"
   | "vault_patch_or_delete_data_source"
-  | "vault_data_source_documents";
+  | "vault_data_source_documents"
```
Comment (philipperolet) [inline front/lib/resources/data_source_resource.ts:55]: Unrelated to this PR; seems a bit surprising to have to create string types for each call site, and track teh origin this way. I wasn't part of the design of this so maybe it's legit but it leaves me wondering; is it the right way to trace calls of resources?

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts [hunk 3]
```diff
@@ -23,9 +23,8 @@ async function handler(
     session,
     req.query.wId as string
   );
-  const owner = auth.workspace();
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts:26]: just to confirm: this is thanks to the new auth wrapper we use on endpoints?

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts [hunk 4]
```diff
@@ -35,6 +34,33 @@ async function handler(
     });
   }
 
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
+    return apiError(req, res, {
+      status_code: 404,
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts:40]: nit: 400

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts [hunk 5]
```diff
@@ -35,6 +34,33 @@ async function handler(
...
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts:51]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/documents/index.ts [hunk 6]
```diff
@@ -19,33 +19,38 @@ async function handler(
...
-
-  if (!owner || !auth.isDustSuperUser()) {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "user_not_found",
-        message: "Could not find the user.",
+        type: "invalid_request_error",
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/documents/index.ts:42]: nit: status code 400 above

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/documents/index.ts [hunk 7]
```diff
@@ -19,33 +19,38 @@ async function handler(
...
       },
     });
   }
 
-  const dataSource = await getDataSource(auth, name);
-
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/documents/index.ts:51]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/index.ts [hunk 8]
```diff
@@ -24,8 +25,35 @@ async function handler(
...
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
+    return apiError(req, res, {
+      status_code: 404,
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/index.ts:37]: nit: 400

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/index.ts [hunk 9]
```diff
@@ -24,8 +25,35 @@ async function handler(
...
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/index.ts:48]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/managed/permissions.ts [hunk 10]
```diff
@@ -32,7 +30,23 @@ async function handler(
...
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/managed/permissions.ts:47]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/search.ts [hunk 11]
```diff
@@ -17,8 +18,34 @@ async function handler(
...
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/search.ts:45]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/index.ts [hunk 12]
```diff
@@ -212,18 +212,23 @@ async function handler(
...
-  const { name } = req.query;
-  if (typeof name !== "string") {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "data_source_not_found",
-        message: "The data source you requested was not found.",
+        type: "invalid_request_error",
```
Comment (philipperolet) [inline front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/index.ts:220]: nit: 400

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/index.ts [hunk 13]
```diff
@@ -212,18 +212,23 @@ async function handler(
...
+        message: "Invalid path parameters.",
       },
     });
   }
 
-  const dataSource = await getDataSource(auth, name);
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/index.ts:229]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts [hunk 14]
```diff
@@ -74,18 +74,23 @@ async function handler(
...
-  const { name } = req.query;
-  if (typeof name !== "string") {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "data_source_not_found",
-        message: "The data source you requested was not found.",
+        type: "invalid_request_error",
```
Comment (philipperolet) [inline front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts:82]: nit: 400

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts [hunk 15]
```diff
@@ -74,18 +74,23 @@ async function handler(
...
+        message: "Invalid path parameters.",
       },
     });
   }
 
-  const dataSource = await getDataSource(auth, name);
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```
Comment (philipperolet) [inline front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts:91]: nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)

Comment (philipperolet) [reply front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts:91]: will stop marking those below from now on :)

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts (hunk 14 above)
Comment (philipperolet) [reply front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/parents.ts:82]: will stop marking those below from now on :)

File: front/pages/api/v1/w/[wId]/data_sources/[dsId]/tables/csv.ts [hunk 16]
```diff
@@ -0,0 +1,79 @@
+import type { NextApiRequest, NextApiResponse } from "next";
```
Comment (philipperolet) [inline front/pages/api/v1/w/[wId]/data_sources/[dsId]/tables/csv.ts:1]: git diff seems to have missed the rename here; assuming the changes were the same typical ones as above

File: front/pages/api/w/[wId]/data_sources/[dsId]/index.ts [hunk 17]
```diff
@@ -21,19 +21,20 @@ async function handler(
   >,
   auth: Authenticator
 ): Promise<void> {
-  if (!req.query.name || typeof req.query.name !== "string") {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
     return apiError(req, res, {
-      status_code: 404,
+      status_code: 400,
```
Comment (philipperolet) [inline front/pages/api/w/[wId]/data_sources/[dsId]/index.ts:27]: this one was :+1: :)

**Review by philipperolet — COMMENTED** [review]
Looks good :) Left a few nits

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts (hunk 3 above)
Comment (spolu) [reply front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts:26]: yes

File: front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts (hunk 5 above)
Comment (spolu) [reply front/pages/api/poke/workspaces/[wId]/data_sources/[dsId]/config.ts:51]: I will remove all of them and move to fetchById only once name usage is gone

File: front/lib/resources/data_source_resource.ts (hunk 2 above)
Comment (spolu) [reply front/lib/resources/data_source_resource.ts:55]: This is transitory to keep track of usage of [name] across the entire product and will get killed once everything is removed

File: front/components/vaults/VaultWebsiteModal.tsx (hunk 1 above)
Comment (spolu) [reply front/components/vaults/VaultWebsiteModal.tsx:180]: As per the comment the previous implementation didn't make any sense

**Comment by spolu** [comment]
Deploying on front-edge to verify it builds + running final tests

@philipperolet addressed most of your comments :+1:

**Review by philipperolet — APPROVED** [review]

**Pull request merged**